lib/iris/pandas.py

```diff
@@ -518,88 +518,298 @@ def _get_base(array):
         data = data.astype("f").filled(np.nan)
-    elif copy:
-        data = data.copy()
 
-    index = columns = None
-    if cube.coords(dimensions=[0]):
-        index = _as_pandas_coord(cube.coord(dimensions=[0]))
-    if cube.coords(dimensions=[1]):
-        columns = _as_pandas_coord(cube.coord(dimensions=[1]))
+    # Extract dim coord information
+    dim_coord_list = [
+        [
+            cube.coords(dimensions=n, dim_coords=True)[0].name(),
+            _as_pandas_coord(cube.coords(dimensions=n, dim_coords=True)[0]),
+        ]
+        if cube.coords(dimensions=n, dim_coords=True)
+        else ["dim" + str(n), range(cube.shape[n])]
+        for n in range(cube.ndim)
+    ]
+    # Extract separate lists for dim names and dim values
+    coord_names, coords = list(zip(*dim_coord_list))
+    # Make base DataFrame
+    index = pandas.MultiIndex.from_product(coords, names=coord_names)
+    data_frame = pandas.DataFrame(
+        data.ravel(), columns=[cube.name()], index=index
+    )
 
-    data_frame = pandas.DataFrame(data, index, columns)
-    if not copy:
-        _assert_shared(data, data_frame)
+    # Add AuxCoord & scalar coordinate information
+    if add_aux_coord:
+        # Extract aux coord information
+        aux_coord_list = list(
+            chain.from_iterable(
+                [
+                    [
+                        [n, coord]
+                        for coord in cube.coords(
+                            dimensions=n, dim_coords=False
+                        )
+                    ]
+                    for n in _get_dim_combinations(cube.ndim)
+                ]
+            )
+        )
+        for aux_coord_index, aux_coord in aux_coord_list:
+            acoord_df = pandas.DataFrame(
+                aux_coord.points.ravel(),
+                columns=[aux_coord.name()],
+                index=pandas.MultiIndex.from_product(
+                    [coords[i] for i in aux_coord_index],
+                    names=[coord_names[i] for i in aux_coord_index],
+                ),
+            )
+            # Merge to main data frame
+            data_frame = pandas.merge(
+                data_frame, acoord_df, left_index=True, right_index=True
+            )
+
+        # Add scalar coordinate information
+        scalar_coord_list = cube.coords(dimensions=(), dim_coords=False)
+        for scalar_coord in scalar_coord_list:
+            data_frame[scalar_coord.name()] = scalar_coord.points.squeeze()
+
+    # Add global attribute information
+    if add_global_attributes:
+        if isinstance(
+            cube.attributes[global_attribute],
+            pp.STASH,
+        ):
+            data_frame[global_attribute] = str(
+                cube.attributes[global_attribute]
+            )
+        else:
+            data_frame[global_attribute] = cube.attributes[global_attribute]
 
-    return data_frame
+    if copy:
+        return data_frame.reorder_levels(coord_names).sort_index()
```

Comment:
This was because the `pandas.merge` functions that underpins much of the dimension mapping of auxiliary data resorts the MultiIndex.  This is to make sure that the final data_frame index ordering matches the that of the cubes.